web/src/components/index/UserTasks.tsx

```diff
@@ -1,17 +1,55 @@
 import { useAuth } from "@/hooks/useAuth";
 import LoginFallback from "@/components/ui/LoginFallback";
+import Button from "../ui/Button";
+import { FormEvent, useRef } from "react";
+import styles from "./UserTasks.module.css";
 
 export default function UserTasks() {
   const { loading, isAuthed, userData } = useAuth();
+  const dialogRef = useRef<HTMLDialogElement>(null);
 
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const inputElement = event.currentTarget.elements.namedItem("chosenName") as HTMLInputElement;
+    const chosenName = inputElement.value;
+    const dialogElement = dialogRef.current;
+
+    try {
+      const response = await fetch("/profile", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ chosen_name: chosenName }),
+      });
```

Comment:
better to use the codegen-ed [function](https://github.com/ucl-arc-tre/portal/blob/d356730451e81800f7510fc1f449632a3ded310a/web/src/openapi/sdk.gen.ts#L31) for all the type safety goodness cf. https://github.com/ucl-arc-tre/portal/blob/9024182498ed647cbb578499a5154809be65a14d/web/src/hooks/useAuth.tsx#L29